### PR TITLE
CLOUDP-205313: patch dbuser update to support external DB

### LINKED ISSUE #2338: Mongo Atlas cli  -  dbuser update doesn't support external DB (X509 certs users)
**Describe the bug**
We want to update roles to a  X509 cert managed dbuser, which is using $external database. 
It seems to be supported only for users using admin database (SCRAM) , but not for users using $external database.

**To Reproduce**
1. Run: `atlas dbusers update myUser-test  --role backup `
2. See error: 
```
Error: https://cloud.mongodb.com/api/atlas/v2/groups/*******/databaseUsers/admin/myUser-test PATCH: HTTP 404 Not Found (Error code: "USERNAME_NOT_FOUND") Detail: No user with username myUser-test exists. Reason: Not Found. Params: [myUser-test]
```

**Expected behavior**
When we are updating another user that uses admin database we get the expected behavior (user's role updated)
Example with a SCRAM user called "prueba"

``` atlas dbuser update prueba --role backup ```
```json
{
  "awsIAMType": "NONE",
  "databaseName": "admin",
  "groupId": "******",
  "ldapAuthType": "NONE",
  "links": [
    {
      "href": "https://cloud.mongodb.com/api/atlas/v2/groups/*******/databaseUsers/admin/prueba",
      "rel": "self"
    }
  ],
  "oidcAuthType": "NONE",
  "roles": [
    {
      "databaseName": "admin",
      "roleName": "backup"
    }
  ],
  "username": "prueba",
  "x509Type": "NONE"
}
```

Json output  of the atlas dbuser list command with myUser-test that uses external database:

```json
    {
      "awsIAMType": "NONE",
      "databaseName": "$external",
      "groupId": "******",
      "ldapAuthType": "NONE",
      "links": [
        {
          "href": "https://cloud.mongodb.com/api/atlas/v2/groups/*****/databaseUsers/$external/myUser-test",
          "rel": "self"
        }
      ],
      "oidcAuthType": "NONE",
      "roles": [
        {
          "databaseName": "admin",
          "roleName": "backup"
        }
      ],
      "username": "myUser-test",
      "x509Type": "MANAGED"
    }
```

**Desktop (please complete the following information):**
 - OS:  tested locally with MacOs Monterrey , also with a docker container (image ubuntu:20.04)
 - Atlas  (from master) / focal/mongodb-org/7.0

**Additional context**
We can see in the error that  the url of the api always uses the admin database.  
Comparing files updates.go with delete.go or describe.go in the repo https://github.com/mongodb/mongodb-atlas-cli/tree/master/internal/cli/atlas/dbusers, we are seeing that the first one haven't a parameter , `--authDB`,  to specify the database name for the user.

### PR DESCRIPTION
**Pull request by matt-condon**
## Proposed changes

_Jira ticket:_ CLOUDP-205313

**Before fix:**
```

{"detail":"No user with username arn:aws:iam::**********:user/my-test-user exists.","error":404,"errorCode":"USERNAME_NOT_FOUND","parameters":["arn:aws:iam::**********:user/my-test-user"],"reason":"Not Found"}
Error: https://cloud-dev.mongodb.com/api/atlas/v2/groups/**********/databaseUsers/admin/arn:aws:iam::**********:user%2Fmy-test-user PATCH: HTTP 404 Not Found (Error code: "USERNAME_NOT_FOUND") Detail: No user with username arn:aws:iam::**********:user/my-test-user exists. Reason: Not Found. Params: [arn:aws:iam::**********:user/my-test-user]
Exiting.
```

**After fix:** 

```
{"awsIAMType":"USER","databaseName":"$external","groupId":"**********","labels":[],"ldapAuthType":"NONE","links":[{"href":"https://cloud.mongodb.com/api/atlas/v2/groups/**********/databaseUsers/$external/arn:aws:iam::**********:user%2Fmy-test-user","rel":"self"}],"oidcAuthType":"NONE","roles":[{"databaseName":"admin","roleName":"readWrite"}],"scopes":[],"username":"arn:aws:iam::**********:user/my-test-user","x509Type":"NONE"}
Successfully updated database user 'arn:aws:iam::**********:user/my-test-user'.
```

Closes #2338

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/dbusers/update.go [hunk 1]
```diff
@@ -79,13 +81,23 @@ func (opts *UpdateOpts) update(out *admin.CloudDatabaseUser) {
...
 	out.Roles = convert.BuildAtlasRoles(opts.roles)
-	out.DatabaseName = convert.GetAuthDB(out)
+	out.DatabaseName = opts.authDB
+	if opts.authDB == "" {
+		out.DatabaseName = convert.GetAuthDB(out)
+	}
+}
+
+func (opts *UpdateOpts) validateAuthDB() error {
+	if opts.authDB == "" {
```
Comment (blva) [inline internal/cli/atlas/dbusers/update.go:93]: [q] what does the API return if you don't set this value?

File: internal/cli/atlas/dbusers/update.go [hunk 2]
```diff
@@ -118,6 +131,7 @@ func UpdateBuilder() *cobra.Command {
 
 	cmd.Flags().StringVarP(&opts.username, flag.Username, flag.UsernameShort, "", usage.DBUsername)
 	cmd.Flags().StringVarP(&opts.password, flag.Password, flag.PasswordShort, "", usage.DBUserPassword)
+	cmd.Flags().StringVar(&opts.authDB, flag.AuthDB, "", usage.AtlasAuthDB)
```
Comment (blva) [inline internal/cli/atlas/dbusers/update.go:134]: should we set admin as default here?

**Review by blva — CHANGES_REQUESTED** [review]
One minor ask if possible, otherwise I can approve

File: internal/cli/atlas/dbusers/update.go (hunk 2 above)
Comment (matt-condon) [reply internal/cli/atlas/dbusers/update.go:134]: Same consideration as above

File: internal/cli/atlas/dbusers/update.go (hunk 1 above)
Comment (blva) [reply internal/cli/atlas/dbusers/update.go:93]: okay, works for me

**Review by blva — APPROVED** [review]

**Review by davidhou17 — APPROVED** [review]
LGTM

**Pull request merged**